src/module/service.cpp

```diff
@@ -129,6 +129,35 @@ ::grpc::Status ModuleService_::ReconfigureResource(
     return grpc::Status();
 };
 
+::grpc::Status ModuleService_::ValidateConfig(
+    ::grpc::ServerContext* context,
+    const ::viam::module::v1::ValidateConfigRequest* request,
+    ::viam::module::v1::ValidateConfigResponse* response) {
+    viam::app::v1::ComponentConfig proto = request->config();
+    Resource cfg = Resource::from_proto(proto);
+
+    std::shared_ptr<ModelRegistration> reg = Registry::lookup_resource(cfg.api, cfg.model);
+    if (reg == nullptr) {
+        return grpc::Status(grpc::UNKNOWN,
+                            "unable to validate resource " + cfg.resource_name().name +
+                                " as it hasn't been registered.");
+    }
+    try {
+        std::vector<std::string> implicit_deps = reg->validate(cfg);
+        for (int i = 0; i < implicit_deps.size(); i++) {
+            if (i == 0) {
+              response->add_dependencies();
+            }
+            response->set_dependencies(i, implicit_deps[i]);
+        }
+    } catch (std::exception& exc) {
```

Comment:
This `catch` is supposed to catch thrown errors in `validate` functions. I tested it by modifying the `validate` function in `example_module` to return an error. Instead of erroring within the CPP SDK, the error is sent back to the parent via gRPC (the intended behavior).

Reply:
Great, thanks for clarifying!